### PR TITLE
Addition of cutt.ly shortening service 

### PR DESCRIPTION
**Pull request by VarthanV**
I have added the implementation of cutt.ly service. It too gives us the analytics of the link that e shortened such as No of Clicks, No of  Visits from Social Networking Sites etc. I have implemented the "short" method. If the author asks to proceed I will implement the analytics function too. Thanks in Advance !

### CODE REVIEW

File: pyshorteners/base.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def _post(self, url, data=None, json=None, params=None, headers=None):
         )
         return response
 
-    def short(self, url):
+    def short(self, url ,api_key =None ):
```
Comment (ellisonleao) [inline pyshorteners/base.py]: `api_key`  is provided upon Shortener object instantion, becoming an object property (`self.api_key`) . Can you remove this kwarg please?

File: pyshorteners/shorteners/cuttly.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class Shortener(BaseShortener):
+
+    """
+    Implementation of https://cutt.ly/ URL shortner
+    Params Needed :API key
+
+    
+    They too provide analyitcs for the URL
```
Comment (ellisonleao) [inline pyshorteners/shorteners/cuttly.py]: typo on "analytics"

File: pyshorteners/shorteners/cuttly.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    
+    They too provide analyitcs for the URL 
+    that you shortened Features to get analytics will be implemented when the author
+    approves this PR
+    """
+
+    api_url = 'https://cutt.ly/api/api.php '
+
+    def short(self, url, api_key=None):
+            if api_key is None:
```
Comment (ellisonleao) [inline pyshorteners/shorteners/cuttly.py]: shoud use `self.api_key` here

File: pyshorteners/shorteners/cuttly.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                return "Please provide a Valid API key"
```
Comment (ellisonleao) [inline pyshorteners/shorteners/cuttly.py]: Shoud raise [`ShorteningErrorException`](https://github.com/ellisonleao/pyshorteners/blob/master/pyshorteners/exceptions.py#L1) here

File: pyshorteners/base.py (hunk 1 above)
Comment (VarthanV) [reply pyshorteners/base.py]: I have removed it as per your advice sir

File: pyshorteners/shorteners/cuttly.py (hunk 2 above)
Comment (VarthanV) [reply pyshorteners/shorteners/cuttly.py]: Changed the spelling

File: pyshorteners/shorteners/cuttly.py (hunk 4 above)
Comment (VarthanV) [reply pyshorteners/shorteners/cuttly.py]: Raised Shortening Error Exception

**Comment by VarthanV** [comment]
Thanks, sir for taking time to review my work. I have made the following changes as per your advice, The changes are
        

- Added Unit Tests.

- Removed the API Key Argument from the "short" method in base.py and started using the args from the Shortener Object Ref: Commit cb1626a.

- Raised Shortening Error Exception at  line num 33

File: pyshorteners/shorteners/cuttly.py [hunk 5]
```diff
@@ -10,32 +10,31 @@ class Shortener(BaseShortener):
...
     """
 
     api_url = 'https://cutt.ly/api/api.php '
 
-    def short(self, url, api_key=None):
-            if api_key is None:
-                return "Please provide a Valid API key"
+    def short(self, url):
+            if self.api_key is None:
+                return BadAPIResponseException("Please provide a valid API Key")
```
Comment (ellisonleao) [inline pyshorteners/shorteners/cuttly.py]: should call `raise` instead of `return`

**Comment by VarthanV** [comment]
Sir I have made the necessary changes but my pull Request was not merged can u just pls guide me in this

**Pull request merged**